<?xml version="1.0" encoding="utf-8"?>
<TestRecord xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Results>
    <UnitTestResult>
      <TestDate>2013-11-03T17:52:11</TestDate>
      <Passed>4</Passed>
      <Errors>1</Errors>
      <Failures>0</Failures>
      <Inconclusive>0</Inconclusive>
      <NotRunnable>0</NotRunnable>
      <Skipped>0</Skipped>
      <Ignored>0</Ignored>
      <Time />
      <Message>One or more child tests had errors</Message>
    </UnitTestResult>
    <UnitTestResult>
      <TestDate>2013-11-03T18:09:57</TestDate>
      <Passed>0</Passed>
      <Errors>5</Errors>
      <Failures>0</Failures>
      <Inconclusive>0</Inconclusive>
      <NotRunnable>0</NotRunnable>
      <Skipped>0</Skipped>
      <Ignored>0</Ignored>
      <Time />
      <Message>One or more child tests had errors</Message>
    </UnitTestResult>
    <UnitTestResult>
      <TestDate>2013-11-03T18:10:27</TestDate>
      <Passed>0</Passed>
      <Errors>0</Errors>
      <Failures>0</Failures>
      <Inconclusive>0</Inconclusive>
      <NotRunnable>0</NotRunnable>
      <Skipped>0</Skipped>
      <Ignored>0</Ignored>
      <Time />
      <Message>Object reference not set to an instance of an object</Message>
      <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
    </UnitTestResult>
    <UnitTestResult>
      <TestDate>2013-11-03T18:12:52</TestDate>
      <Passed>0</Passed>
      <Errors>5</Errors>
      <Failures>0</Failures>
      <Inconclusive>0</Inconclusive>
      <NotRunnable>0</NotRunnable>
      <Skipped>0</Skipped>
      <Ignored>0</Ignored>
      <Time />
      <Message>One or more child tests had errors</Message>
    </UnitTestResult>
    <UnitTestResult>
      <TestDate>2013-11-03T18:14:12</TestDate>
      <Passed>0</Passed>
      <Errors>5</Errors>
      <Failures>0</Failures>
      <Inconclusive>0</Inconclusive>
      <NotRunnable>0</NotRunnable>
      <Skipped>0</Skipped>
      <Ignored>0</Ignored>
      <Time />
      <Message>One or more child tests had errors</Message>
    </UnitTestResult>
    <UnitTestResult>
      <TestDate>2013-11-03T18:20:12</TestDate>
      <Passed>4</Passed>
      <Errors>1</Errors>
      <Failures>0</Failures>
      <Inconclusive>0</Inconclusive>
      <NotRunnable>0</NotRunnable>
      <Skipped>0</Skipped>
      <Ignored>0</Ignored>
      <Time />
      <Message>One or more child tests had errors</Message>
    </UnitTestResult>
    <UnitTestResult>
      <TestDate>2013-11-03T18:21:05</TestDate>
      <Passed>4</Passed>
      <Errors>1</Errors>
      <Failures>0</Failures>
      <Inconclusive>0</Inconclusive>
      <NotRunnable>0</NotRunnable>
      <Skipped>0</Skipped>
      <Ignored>0</Ignored>
      <Time />
      <Message>One or more child tests had errors</Message>
    </UnitTestResult>
  </Results>
  <Tests>
    <TestRecord Name="MarsRoverTest">
      <Results>
        <UnitTestResult>
          <TestDate>2013-11-03T17:52:11</TestDate>
          <Passed>4</Passed>
          <Errors>1</Errors>
          <Failures>0</Failures>
          <Inconclusive>0</Inconclusive>
          <NotRunnable>0</NotRunnable>
          <Skipped>0</Skipped>
          <Ignored>0</Ignored>
          <Time />
          <Message>One or more child tests had errors</Message>
        </UnitTestResult>
        <UnitTestResult>
          <TestDate>2013-11-03T18:09:57</TestDate>
          <Passed>0</Passed>
          <Errors>5</Errors>
          <Failures>0</Failures>
          <Inconclusive>0</Inconclusive>
          <NotRunnable>0</NotRunnable>
          <Skipped>0</Skipped>
          <Ignored>0</Ignored>
          <Time />
          <Message>One or more child tests had errors</Message>
        </UnitTestResult>
        <UnitTestResult>
          <TestDate>2013-11-03T18:12:52</TestDate>
          <Passed>0</Passed>
          <Errors>5</Errors>
          <Failures>0</Failures>
          <Inconclusive>0</Inconclusive>
          <NotRunnable>0</NotRunnable>
          <Skipped>0</Skipped>
          <Ignored>0</Ignored>
          <Time />
          <Message>One or more child tests had errors</Message>
        </UnitTestResult>
        <UnitTestResult>
          <TestDate>2013-11-03T18:14:12</TestDate>
          <Passed>0</Passed>
          <Errors>5</Errors>
          <Failures>0</Failures>
          <Inconclusive>0</Inconclusive>
          <NotRunnable>0</NotRunnable>
          <Skipped>0</Skipped>
          <Ignored>0</Ignored>
          <Time />
          <Message>One or more child tests had errors</Message>
        </UnitTestResult>
        <UnitTestResult>
          <TestDate>2013-11-03T18:20:12</TestDate>
          <Passed>4</Passed>
          <Errors>1</Errors>
          <Failures>0</Failures>
          <Inconclusive>0</Inconclusive>
          <NotRunnable>0</NotRunnable>
          <Skipped>0</Skipped>
          <Ignored>0</Ignored>
          <Time />
          <Message>One or more child tests had errors</Message>
        </UnitTestResult>
        <UnitTestResult>
          <TestDate>2013-11-03T18:21:05</TestDate>
          <Passed>4</Passed>
          <Errors>1</Errors>
          <Failures>0</Failures>
          <Inconclusive>0</Inconclusive>
          <NotRunnable>0</NotRunnable>
          <Skipped>0</Skipped>
          <Ignored>0</Ignored>
          <Time />
          <Message>One or more child tests had errors</Message>
        </UnitTestResult>
      </Results>
      <Tests>
        <TestRecord Name="Test">
          <Results>
            <UnitTestResult>
              <TestDate>2013-11-03T17:52:11</TestDate>
              <Passed>4</Passed>
              <Errors>1</Errors>
              <Failures>0</Failures>
              <Inconclusive>0</Inconclusive>
              <NotRunnable>0</NotRunnable>
              <Skipped>0</Skipped>
              <Ignored>0</Ignored>
              <Time />
              <Message>One or more child tests had errors</Message>
            </UnitTestResult>
            <UnitTestResult>
              <TestDate>2013-11-03T18:09:57</TestDate>
              <Passed>0</Passed>
              <Errors>5</Errors>
              <Failures>0</Failures>
              <Inconclusive>0</Inconclusive>
              <NotRunnable>0</NotRunnable>
              <Skipped>0</Skipped>
              <Ignored>0</Ignored>
              <Time />
              <Message>One or more child tests had errors</Message>
            </UnitTestResult>
            <UnitTestResult>
              <TestDate>2013-11-03T18:12:52</TestDate>
              <Passed>0</Passed>
              <Errors>5</Errors>
              <Failures>0</Failures>
              <Inconclusive>0</Inconclusive>
              <NotRunnable>0</NotRunnable>
              <Skipped>0</Skipped>
              <Ignored>0</Ignored>
              <Time />
              <Message>One or more child tests had errors</Message>
            </UnitTestResult>
            <UnitTestResult>
              <TestDate>2013-11-03T18:14:12</TestDate>
              <Passed>0</Passed>
              <Errors>5</Errors>
              <Failures>0</Failures>
              <Inconclusive>0</Inconclusive>
              <NotRunnable>0</NotRunnable>
              <Skipped>0</Skipped>
              <Ignored>0</Ignored>
              <Time />
              <Message>One or more child tests had errors</Message>
            </UnitTestResult>
            <UnitTestResult>
              <TestDate>2013-11-03T18:20:12</TestDate>
              <Passed>4</Passed>
              <Errors>1</Errors>
              <Failures>0</Failures>
              <Inconclusive>0</Inconclusive>
              <NotRunnable>0</NotRunnable>
              <Skipped>0</Skipped>
              <Ignored>0</Ignored>
              <Time />
              <Message>One or more child tests had errors</Message>
            </UnitTestResult>
            <UnitTestResult>
              <TestDate>2013-11-03T18:21:05</TestDate>
              <Passed>4</Passed>
              <Errors>1</Errors>
              <Failures>0</Failures>
              <Inconclusive>0</Inconclusive>
              <NotRunnable>0</NotRunnable>
              <Skipped>0</Skipped>
              <Ignored>0</Ignored>
              <Time />
              <Message>One or more child tests had errors</Message>
            </UnitTestResult>
          </Results>
          <Tests>
            <TestRecord Name="ShouldMoveBackwardOneSecond">
              <Results>
                <UnitTestResult>
                  <TestDate>2013-11-03T17:52:11</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.28ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:09:57</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveBackwardOneSecond () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:42 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:10:41</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Object reference not set to an instance of an object</Message>
                  <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:12:23</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Object reference not set to an instance of an object</Message>
                  <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:12:52</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveBackwardOneSecond () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:42 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:13:00</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveBackwardOneSecond () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:42 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:13:09</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Object reference not set to an instance of an object</Message>
                  <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:14:12</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>System.NullReferenceException : Object reference not set to an instance of an object</Message>
                  <StackTrace>  at MarsRover.Robot.ProcessCommand (MarsRover.Movement command) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRover.Robot.ProcessCommands (IList`1 commands) [0x00013] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:51 
  at MarsRover.Robot.Move (System.String[] movements) [0x00008] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:23 
  at MarsRoverTest.Test.ShouldMoveBackwardOneSecond () [0x00039] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:41 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:15:38</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>System.NullReferenceException : Object reference not set to an instance of an object</Message>
                  <StackTrace>  at MarsRover.Robot.ProcessCommand (MarsRover.Movement command) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRover.Robot.ProcessCommands (IList`1 commands) [0x00013] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:51 
  at MarsRover.Robot.Move (System.String[] movements) [0x00008] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:23 
  at MarsRoverTest.Test.ShouldMoveBackwardOneSecond () [0x00039] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:41 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:15:53</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Object reference not set to an instance of an object</Message>
                  <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:15:46</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Couldn't create a remote process.</Message>
                  <StackTrace>  at MonoDevelop.Core.Execution.ProcessHostController.CreateInstance (System.String assemblyPath, System.String typeName, System.String[] addins, IList`1 userAssemblyPaths) [0x00000] in &lt;filename unknown&gt;:0 
  at (wrapper remoting-invoke-with-check) MonoDevelop.Core.Execution.ProcessHostController:CreateInstance (string,string,string[],System.Collections.Generic.IList`1&lt;string&gt;)
  at MonoDevelop.Core.Execution.ProcessService.CreateExternalProcessObject (System.Type type, IExecutionHandler executionHandler, IList`1 userAssemblyPaths) [0x00000] in &lt;filename unknown&gt;:0 
  at MonoDevelop.NUnit.NUnitAssemblyTestSuite.RunUnitTest (MonoDevelop.NUnit.UnitTest test, System.String suiteName, System.String pathName, System.String testName, MonoDevelop.NUnit.TestContext testContext) [0x00000] in &lt;filename unknown&gt;:0 
  at MonoDevelop.NUnit.NUnitTestCase.OnRun (MonoDevelop.NUnit.TestContext testContext) [0x00000] in &lt;filename unknown&gt;:0 
  at MonoDevelop.NUnit.UnitTest.Run (MonoDevelop.NUnit.TestContext testContext) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:16:15</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Object reference not set to an instance of an object</Message>
                  <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:16:23</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Object reference not set to an instance of an object</Message>
                  <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:20:12</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.29ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:21:05</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.28ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
              </Results>
            </TestRecord>
            <TestRecord Name="ShouldMoveFordWardThreeSeconds">
              <Results>
                <UnitTestResult>
                  <TestDate>2013-11-03T17:52:11</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 3))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFordWardThreeSeconds () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:72 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:09:57</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 3))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFordWardThreeSeconds () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:72 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:12:52</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 3))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFordWardThreeSeconds () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:72 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:14:12</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>System.NullReferenceException : Object reference not set to an instance of an object</Message>
                  <StackTrace>  at MarsRover.Robot.ProcessCommand (MarsRover.Movement command) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRover.Robot.ProcessCommands (IList`1 commands) [0x00013] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:51 
  at MarsRover.Robot.Move (System.String[] movements) [0x00008] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:23 
  at MarsRoverTest.Test.ShouldMoveFordWardThreeSeconds () [0x00039] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:71 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:20:12</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 3))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFordWardThreeSeconds () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:72 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:21:05</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 3))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFordWardThreeSeconds () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:72 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:22:10</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 3))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFordWardThreeSeconds () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:72 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:24:10</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Object reference not set to an instance of an object</Message>
                  <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:24:38</TestDate>
                  <Passed>0</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Object reference not set to an instance of an object</Message>
                  <StackTrace>  at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke (System.Runtime.Remoting.Proxies.RealProxy rp, IMessage msg, System.Exception&amp; exc, System.Object[]&amp; out_args) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:24:46</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 3))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFordWardThreeSeconds () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:72 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
              </Results>
            </TestRecord>
            <TestRecord Name="ShouldMoveFowardOneSecond">
              <Results>
                <UnitTestResult>
                  <TestDate>2013-11-03T17:52:11</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:09:57</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Forward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFowardOneSecond () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:32 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:12:52</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Forward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveFowardOneSecond () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:32 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:14:12</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>System.NullReferenceException : Object reference not set to an instance of an object</Message>
                  <StackTrace>  at MarsRover.Robot.ProcessCommand (MarsRover.Movement command) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRover.Robot.ProcessCommands (IList`1 commands) [0x00013] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:51 
  at MarsRover.Robot.Move (System.String[] movements) [0x00008] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:23 
  at MarsRoverTest.Test.ShouldMoveFowardOneSecond () [0x00039] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:31 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:20:12</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:21:05</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
              </Results>
            </TestRecord>
            <TestRecord Name="ShouldMoveLeftOneSecond">
              <Results>
                <UnitTestResult>
                  <TestDate>2013-11-03T17:52:11</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:09:57</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveLeftOneSecond () [0x000e4] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:63 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:12:52</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveLeftOneSecond () [0x000e4] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:63 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:14:12</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>System.NullReferenceException : Object reference not set to an instance of an object</Message>
                  <StackTrace>  at MarsRover.Robot.ProcessCommand (MarsRover.Movement command) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRover.Robot.ProcessCommands (IList`1 commands) [0x00013] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:51 
  at MarsRover.Robot.Move (System.String[] movements) [0x00008] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:23 
  at MarsRoverTest.Test.ShouldMoveLeftOneSecond () [0x00039] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:61 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:20:12</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:21:05</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
              </Results>
            </TestRecord>
            <TestRecord Name="ShouldMoveRightOneSecond">
              <Results>
                <UnitTestResult>
                  <TestDate>2013-11-03T17:52:11</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:09:57</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveRightOneSecond () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:52 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:12:52</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Moq.MockException : 
Expected invocation on the mock at least once, but was never performed: x =&gt; x.Backward(It.Is&lt;Int32&gt;(y =&gt; y == 1))
No setups configured.
No invocations performed.</Message>
                  <StackTrace>  at Moq.Mock.ThrowVerifyException (Moq.MethodCall expected, IEnumerable`1 setups, IEnumerable`1 actualCalls, System.Linq.Expressions.Expression expression, Times times, Int32 callCount) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.VerifyCalls (Moq.Interceptor targetInterceptor, Moq.MethodCall expected, System.Linq.Expressions.Expression expression, Times times) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock.Verify[IEngine] (Moq.Mock mock, System.Linq.Expressions.Expression`1 expression, Times times, System.String failMessage) [0x00000] in &lt;filename unknown&gt;:0 
  at Moq.Mock`1[MarsRover.IEngine].Verify (System.Linq.Expressions.Expression`1 expression) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRoverTest.Test.ShouldMoveRightOneSecond () [0x00045] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:52 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:14:12</TestDate>
                  <Passed>0</Passed>
                  <Errors>1</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>System.NullReferenceException : Object reference not set to an instance of an object</Message>
                  <StackTrace>  at MarsRover.Robot.ProcessCommand (MarsRover.Movement command) [0x00000] in &lt;filename unknown&gt;:0 
  at MarsRover.Robot.ProcessCommands (IList`1 commands) [0x00013] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:51 
  at MarsRover.Robot.Move (System.String[] movements) [0x00008] in /home/luis/tmp/MarsRover/MarsRover/Robot.cs:23 
  at MarsRoverTest.Test.ShouldMoveRightOneSecond () [0x00039] in /home/luis/tmp/MarsRover/MarsRoverTest/Test.cs:51 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&amp;)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in &lt;filename unknown&gt;:0 </StackTrace>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:20:12</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
                <UnitTestResult>
                  <TestDate>2013-11-03T18:21:05</TestDate>
                  <Passed>1</Passed>
                  <Errors>0</Errors>
                  <Failures>0</Failures>
                  <Inconclusive>0</Inconclusive>
                  <NotRunnable>0</NotRunnable>
                  <Skipped>0</Skipped>
                  <Ignored>0</Ignored>
                  <Time />
                  <Message>Test successful

Execution time: 0.00ms</Message>
                  <ConsoleOutput />
                  <ConsoleError />
                </UnitTestResult>
              </Results>
            </TestRecord>
          </Tests>
        </TestRecord>
      </Tests>
    </TestRecord>
  </Tests>
</TestRecord>